SPECS/libcriterion/libcriterion.spec

```diff
@@ -0,0 +1,57 @@
+%define pr_name criterion
+Summary:        A cross-platform C and C++ unit testing framework for the 21st century
+Name:           lib%{pr_name}
+Version:        2.4.1
```

Comment:
Hey Bala...I see that this package bundles quite a few other open source packages (rather than linking against them).   Many other packages in Mariner have vendored components too, so it's not unheard of, but it's very undesirable.  Are there any build options here?  Can we add the individual packages and change libcriterion to dynamically link against these?  The problem with vendoring is that it's harder to find the CVE's and we need to fix them in multiple places (e.g.  libffi is needed here...if libffi has a CVE in it we have to fix this code and the actual libffi)

[boxfort](https://github.com/diacritic/BoxFort)
[debugbreak](https://github.com/scottt/debugbreak)
[libffi](https://sourceware.org/libffi/)
[klib](http://attractivechaos.github.io/klib/)
[libcsptr](https://github.com/Snaipe/libcsptr)
[nanomsg](http://nanomsg.org/)
[nanopb](http://koti.kapsi.fi/jpa/nanopb/)